library/core/src/panic.rs

```diff
@@ -90,6 +90,27 @@ pub macro unreachable_2021 {
     ),
 }
 
+#[doc(hidden)]
+#[unstable(feature = "core_panic", issue = "none")]
+#[allow_internal_unstable(core_panic, const_format_args)]
+#[rustc_macro_transparency = "semitransparent"]
+pub macro debug_assert_nounwind {
```

Comment:
This macro should have documentation that explains that it should be used in unsafe contexts because it cannot compromise unwind safety.

Reply:
Well, actually no unwind safety is compromised and `debug_assert!` can be used in these contexts, because if unwinding would only happen when there's an UB, and unwinding happening at undesirable place is an allowed behaviour for UB.

That said, causing UB by unwinding certainly defeats the purpose of having these assertions for debugging.